wsmaster/che-core-api-factory-gitlab/src/main/java/org/eclipse/che/api/factory/server/gitlab/GitlabUrl.java

```diff
@@ -185,15 +183,22 @@ public String location() {
    * @return location of specified file in a repository
    */
   public String rawFileLocation(String fileName) {
-    StringJoiner joiner = new StringJoiner("/").add(hostName).add(username).add(project);
-    if (repository != null) {
-      joiner.add(repository);
-    }
-    joiner.add("-").add("raw").add(firstNonNull(branch, "master"));
-    if (subfolder != null) {
-      joiner.add(subfolder);
+    StringJoiner joiner =
+        new StringJoiner("/")
+            .add(hostName)
+            .add("api/v4/projects")
+            .add(username + "%2f" + project)
```

Comment:
why this needs to be encoded as `%2f` ?

Reply:
It's requirement of API:
![image](https://user-images.githubusercontent.com/1651062/112297048-16318e80-8c9e-11eb-8fab-a85e36688837.png)